### ISSUE TITLE
TrackBy for arrays

### ISSUE THREAD

**Issue by let4be** [d4l3k/messagediff#15]
it would be interesting if we could define a field by which we track objects inside array(similar as angular.js or vue.js)
This way we could precisely tell which objects were deleted/added to an array when doing a diff

**Comment by d4l3k** [comment]
Yeah, that's definitely possible. I'd welcome a PR to do that, or some sort of LCS detection.
